### PR TITLE
Add support to build a wheel with libs on Linux and macOS

### PR DESCRIPTION
**Pull request by daniviga**
Thanks to some techniques like `RPATH` (via Cmake) and to a lot of magic we are able to build a reproducible and portable rtree wheel for macOS and Linux. It includes libspatialindex.

### CODE REVIEW

**Comment by micheles** [comment]
25 commits. It must be right!
LGTM

File: setup.py [hunk 1]
```diff
@@ -1,52 +1,146 @@
...
+    tar = tarfile.open(archive)
+    tar.extractall()
+    tar.close()
+
+
+# attempt to run parallel make with at most 8 workers
+def cpu_count():
+    try:
+        import multiprocessing
+        return min(8, multiprocessing.cpu_count())
```
Comment (micheles) [inline setup.py]: Why the try except?

Also, in Python 3, you should return `os.cpu_count()`

File: setup.py [hunk 2]
```diff
@@ -1,52 +1,146 @@
...
+if __name__ == "__main__":
+    if os.name == 'nt':
+        data_files = [('Lib/site-packages/rtree',
+                      [os.environ['SPATIALINDEX_LIBRARY']
+                          if 'SPATIALINDEX_LIBRARY' in os.environ else
+                          r'D:\libspatialindex\bin\spatialindex.dll',
+                       os.environ['SPATIALINDEX_C_LIBRARY']
+                          if 'SPATIALINDEX_C_LIBRARY' in os.environ else
+                          r'D:\libspatialindex\bin\spatialindex_c.dll'])]
+    else:
```
Comment (micheles) [inline setup.py:106]: Why `D:` ? It is just a convention we are using?

Comment (daniviga) [reply setup.py:106]: This is from upstream. I did not touched it.

File: setup.py (hunk 1 above)
Comment (daniviga) [reply setup.py]: Copy / paste from GDAL :) I would like anyway to keep Python 2 backward compatibility. I'll remove the try: except:

**Pull request merged**